Fri May 30 04:17:02 2008 UTC (5 weeks, 3 days ago) by plougher:

Fix nasty bug in get_dir_index_using_offset().  Thanks to Geert Uytterhoeven
for spotting this!

See also:
http://sourceforge.net/mailarchive/forum.php?thread_name=Pine.LNX.4.64.0805291610580.3218%40vixen.sonytel.be&forum_name=squashfs-devel

diff -p -up linux-2.6.25/fs/squashfs/inode.c.orig linux-2.6.25/fs/squashfs/inode.c
--- linux-2.6.25/fs/squashfs/inode.c.orig	2008-07-08 15:53:26.000000000 -0400
+++ linux-2.6.25/fs/squashfs/inode.c	2008-07-08 15:55:25.000000000 -0400
@@ -1786,7 +1786,7 @@ static int get_dir_index_using_offset(st
 	TRACE("Entered get_dir_index_using_offset, i_count %d, f_pos %d\n",
 					i_count, (unsigned int) f_pos);
 
-	f_pos =- 3;
+	f_pos -= 3;
 	if (f_pos == 0)
 		goto finish;
 
